frontend/vite-project/src/components/card-component/ProductCard.tsx

```diff
@@ -0,0 +1,62 @@
+import { useState, useEffect } from 'react';
+import { ProductCardProps } from '../../types/types';
+import { QuantityControl } from '../quantity-control/QuantityControl';
+import './productCard.css';
+
+const ProductCard: React.FC<ProductCardProps> = ({
+  imageUrl,
+  seedName,
+  price,
+  quantity,
+  onQuantityChange,
+  onAddToCart
+}) => {
+  const [counter, setCounter] = useState(quantity);
+
+  useEffect(() => {
+    setCounter(quantity);
+  }, [quantity]);
+
+  const handleDecrement = () => {
+    const newCount = Math.max(counter - 1, 0);
+    setCounter(newCount);
+    onQuantityChange?.(newCount);
+  };
+
+  const handleIncrement = () => {
+    const newCount = counter + 1;
+    setCounter((prevState) => (prevState -1 === 0 ? 1: prevState -1));
+    onQuantityChange?.(newCount);
+  };
```

Comment:
```suggestion
  const handleDecrement = () => {
    setCounter((prevState) => (prevState -1 === 0 ? 1: prevState -1));
    onQuantityChange?.(counter);
  };

  const handleIncrement = () => {
    const newCount = counter + 1;
     setCounter(newCount);
    onQuantityChange?.(newCount);
  };
```
 The change I suggested was for the handleDecrement function. Perhaps I added the comment on the wrong line before.

Reply:
Eagle eye both of you :)
I have already fixed this issue, the increment/decrement works ;)
I have to pull from main and push :)